### PR TITLE
Support HTTP DELETE

### PR DESCRIPTION
**Pull request by justin-r-oliver**

### CODE REVIEW

File: test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/RestClientTest.java [hunk 1]
```diff
@@ -94,6 +97,19 @@ public void shouldSendQueryWithHeaders() {
...
 
+    @Test
+    public void shouldSendDelete() {
+
+        stubFor(delete(urlEqualTo(RESOURCE_PATH))
+                .withHeader(CONTENT_TYPE, equalTo(CONTENT_TYPE_VALUE))
+                .withHeader(HEADER, equalTo(HEADER_VALUE))
+                .willReturn(aResponse().withStatus(NO_CONTENT_CODE)));
+
+        Response response = restClient.deleteCommand(URL, CONTENT_TYPE_VALUE, getHeaders());
```
Comment (mapingo) [inline test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/RestClientTest.java]: response can be final

File: test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/RestClientTest.java [hunk 2]
```diff
@@ -94,6 +97,19 @@ public void shouldSendQueryWithHeaders() {
...
+
+        assertThat(response.getStatus(), org.hamcrest.CoreMatchers.equalTo(NO_CONTENT_CODE));
```
Comment (mapingo) [inline test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/RestClientTest.java]: CoreMatchers can be static imported

**Review by mapingo — CHANGES_REQUESTED** [review]

File: test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/RestClientTest.java (hunk 2 above)
Comment (justin-r-oliver) [reply test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/RestClientTest.java]: The static import of CoreMatchers.equalTo would conflict with the existing static import of Wiremock.equalTo

File: test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/RestClientTest.java (hunk 1 above)
Comment (justin-r-oliver) [reply test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/RestClientTest.java]: Done

**Review by mapingo — APPROVED** [review]

**Pull request merged**